src/ZipFile.jl

```diff
@@ -141,45 +141,78 @@ type WritableFile <: IO
 	end
 end
 
-type Writer
-	_io :: IO
-	files :: Vector{WritableFile} # files (being) written
-	_current :: Union(WritableFile, Nothing)
-	_closed :: Bool
+if VERSION > v"0.3.11" then
+	type Writer
+		_io :: IO
+		files :: Vector{WritableFile} # files (being) written
+		_current :: Union{WritableFile, Void}
```

Comment:
If these only differ in `Union()` vs `Union{}`, you should use `@compat(Union{})` and it will work on both 0.3 and 0.4, since there is a `using Compat` above.